### PR TITLE
Gh 2270 JUnit5 upgrade - module: example

### PR DESCRIPTION
**Pull request by macenturalxl1**
Hi - I've fixed the code coverage in this Pull Request for the core and example modules - I had to raise another PR so it could generate a code coverage report and point out the classes that had reduced coverage - let me know what you think

### CODE REVIEW

**Comment by macenturalxl1** [comment]
Hi - I've made the changes to ensure the code is covered properly so ready to review / then merge.

Can merge this Pull Request 2nd after this one https://github.com/gchq/Gaffer/pull/2277  

This branch has the rest of the 'core' modules changes so can merge this one in also and then close https://github.com/gchq/Gaffer/pull/2277

File: core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java [hunk 1]
```diff
@@ -414,13 +391,11 @@ public void shouldValidateParentSchemaIdCannotBeUsedWhenGraphIdAlreadyExists() {
...
-            createGraph(export);
-            fail(EXCEPTION_EXPECTED);
-        } catch (final IllegalArgumentException e) {
-            assertEquals("Validation errors: \n" +
-                    String.format(GRAPH_S_ALREADY_EXISTS_SO_YOU_CANNOT_USE_A_DIFFERENT_S_DO_NOT_SET_THE_S_FIELD, "graphId1", SCHEMA_STRING, "parentSchemaIds"), e.getMessage());
-        }
+        final Exception exception = assertThrows(IllegalArgumentException.class, () -> createGraph(export));
+
+        final String expected = "Validation errors: \n" +
+                "Graph: graphId1 already exists so you cannot use a different Schema. Do not set the parentSchemaIds field";
```
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java:397]: Sorry just force of habit isolating tests - realise now that importing the template exceptions doesn't make a difference and actually helps when the exception messages change across multiple classes - I'll change them back to using the imported templates...

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/OperationTest.java [hunk 2]
```diff
@@ -108,10 +109,10 @@ public void shouldHaveSummaryAnnotation() {
 
         // Then
         if (null == annotation || null == annotation.value()) {
-            throw new AssumptionViolatedException("Missing Summary annotation on class " + instance.getClass().getName());
+            throw new AssumptionViolatedException("Missing Since annotation on class " + instance.getClass().getName());
```
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/operation/src/test/java/uk/gov/gchq/gaffer/operation/OperationTest.java:112]: Thanks for spotting the error

File: core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java:397]: Before I change it back - do you have a preference on the imported messages? Would you like me to revert to the original or keep the current change?

[the remark this answers is not in the document]
Comment (macenturalxl1) [reply core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java:397]: Hi just checking if you have a preference on this first? If you prefer to change back to original?

[the remark this answers is not in the document]
Comment (t11947) [reply core/graph/src/test/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherGraphHandlerTest.java:397]: Apologies for the delay in getting back to you. I do see your point about test isolation. Unfortunately, it seems that we've not followed either style consistently throughout the project. I'm happy with the way they are.

**Comment by phosphorbronze50** [comment]
Pull request branch updated to include merge of pull request #2289.

**Comment by t11947** [comment]
Could you merge in the changes made in github.com/gchq/gaffer/issues/2290? I believe the changes related to the issue are in core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java.

**Comment by phosphorbronze50** [comment]
Latest develop merged into the branch for this pull request.
The github.com//issues/2290 comments indicate it was merged into develop before develop changes were pulled to this branch.
Can this be re-reviewed, please?

**Comment by t11947** [comment]
I notice that a lot of the updated JUnit changes have been reverted (use of AssertThrows as an example). Was this necessary for merging in the changes?

**Comment by phosphorbronze50** [comment]
All the code is JUnit5 compliant.
Where there were duplications of the JUnit5 changes already merged into develop these were the ones kept.
There are a couple of places where store integration tests had been partially converted to JUnit 5. These have been kept down to JUnit 4 as there are problems where different parts of the integration test mechanism are on different JUnit versions.

**Review by d47853 — APPROVED** [review]

**Pull request merged**